connect/src/com/telenor/connect/ConnectSdk.java

```diff
@@ -146,22 +135,59 @@ public static synchronized void authenticate(final Activity activity,
                                                  final int customLoadingLayout,
                                                  final int requestCode) {
         Validator.sdkInitialized();
-        sdkProfile.onStartAuthorization(new SdkProfile.OnStartAuthorizationCallback() {
-            @Override
-            public void onSuccess() {
-                Intent intent = getAuthIntent(parameters);
-                intent.putExtra(ConnectUtils.CUSTOM_LOADING_SCREEN_EXTRA, customLoadingLayout);
-                activity.startActivityForResult(intent, requestCode);
-            }
+        final ArrayList<String> uiLocales = getUiLocales();
+        sdkProfile.onStartAuthorization(
+                parameters,
+                uiLocales,
+                new SdkProfile.OnStartAuthorizationCallback() {
 
-            @Override
-            public void onError() {
-                showAuthCancelMessage(activity);
-            }
+                    @Override
+                    public void onSuccess(Uri authorizationStartUri) {
+                        startAuthActivity(authorizationStartUri, false);
+                    }
+
+                    @Override
+                    public void onDivert(Uri diversionUri) {
```

Comment:
I'm guessing the work done per https://github.com/telenordigital/connect-android-sdk/pull/54/files/8514c618036b23ada24c6ba55e45b03c3ad6c0b2#diff-edfffb4e70b7f3f2268239f7a63b2636R43 moves this method somewhere else? If that's the case, consider if the naming could be made less generic.